### PR TITLE
[2.19.x] DDF-5850: Fixed small polygon buffer width exception

### LINKED ISSUE #5850: Openlayers map polygon buffer allows width in the micrometer range
### Description
Current Openlayers (2D) map allows setting arbitrarily small values for buffer sizes. This is not very useful and results in exceptions when performing polygon searches.

### Steps to Reproduce
1. Open Intrigue map view
2. Select `Add Visual -> 2D Map`
3. Click `Search DDF Intrigue`
4. Under `Located` select `Somewhere Specific`
5. From the `Location` drop-down select `Polygon`
6. Click the `Buffer width` input and type `.0` (the input value is auto-filled as `0.000001` which is 1 micrometer). For reference, a human red blood cell is 5-10 micrometers.
7. Click `Draw`
8. Draw 2-3 segment polygon
9. Observe the polygon disappear while attempting to draw another segment

10. Optionally have the `Developer Tools` in Chrome open (or Firefox equivalent) and observe
```
Uncaught Error: Each LinearRing of a Polygon must have 4 or more Positions.
    at Object.module.exports.polygon (index.js?5cbd:84)
    at eval (openlayers.polygon.js?74df:169)
    at Array.map (<anonymous>)
    at eval (openlayers.polygon.js?74df:164)
    at Array.map (<anonymous>)
    at child.drawBorderedPolygon (openlayers.polygon.js?74df:160)
    at eval (openlayers.polygon.js?74df:268)
```
exception in the console

#### Expected behavior :
Valid, buffered-polygon displayed

#### Actual behavior:
Polygon disappears from the screen entirely, without feedback, requiring the user to `Cancel Drawing` and restart

#### Reproduces how often:
Always.

#### Platform and environment:
macOS 10.15.3, Java 1.8.0_202, Chrome 80.0.3987.116

### Affects versions
2.19.3

### Additional Information
Search polygon buffer settings
Screen Shot 2020-02-24 at 10 56 14 AM

### PR DESCRIPTION
**Pull request by leo-sakh**
#### What does this PR do?
Fixed #5850 

#### Who is reviewing it? 
@bennuttle 
@aj-brooks 
@lavoywj 
@AzGoalie 

#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.
@AzGoalie
@bdeining

#### How should this be tested?
See #5850 reproduce steps. These would be the same to verify fix.

#### What are the relevant tickets?
Fixes: #5850 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/js/widgets/openlayers.polygon.js [hunk 1]
```diff
@@ -49,6 +49,19 @@ const translateFromOpenlayersCoordinates = coords => {
...
 
+/**
+ * Returns buffer width in meters with the lowest value of 1 meter.
+ */
+const getConstrainedBufferWidth = model => {
+  const bufferWidth =
+    DistanceUtils.getDistanceInMeters(
+      model.get('polygonBufferWidth'),
+      model.get('polygonBufferUnits')
+    ) || 1
```
Comment (millerw8) [inline ui/packages/catalog-ui-search/src/main/webapp/js/widgets/openlayers.polygon.js]: this doesn't prohibit not using a buffer does it?

**Comment by leo-sakh** [comment]
> Does this mean if a user enters a buffer width less than 1, that we ignore that input and instead use 1 as a baseline?
> 
> If this is the case, is it not a bit confusing for someone? They may see unintended results. Perhaps we should adjust
> 
> https://github.com/codice/ddf/blob/421b1a5a4b9609d48097b67efb03cabaf7debf0a/ui/packages/catalog-ui-search/src/main/webapp/react-component/location/base.line.js#L87
> 
> accordingly?

So `DistanceUtils.getDistanceInMeters` will convert the buffer to meters first and then, yes, this will make sure it's no less than 1. This is what I mentioned during standup, the UI component will be updated to make sure the user sees the feedback.

File: ui/packages/catalog-ui-search/src/main/webapp/js/widgets/openlayers.polygon.js (hunk 1 above)
Comment (leo-sakh) [reply ui/packages/catalog-ui-search/src/main/webapp/js/widgets/openlayers.polygon.js]: No, this only makes sure we do not apply buffers less than 1 meter

**Comment by bdeining** [comment]
build now

**Comment by bdeining** [comment]
verified that drawing works with a small buffer and no errors are in the console

**Comment by leo-sakh** [comment]
build now

**Pull request merged**